### PR TITLE
Run tests on Internet Explorer 11

### LINKED ISSUE #7734: Do we really need latest sinon?
After https://github.com/Leaflet/Leaflet/pull/7725#issuecomment-957769787 we lost the ability to run test suite in IE8-10.
Is it OK?

If it is - then those part of config should be removed.
If it is not - then `sinon` should be downgraded.

Opinions?

### PR DESCRIPTION
**Pull request by jonkoops**
Adds the ability to run the tests on different operating systems and enables test runs for Internet Explorer 11. This PR also downgrades Sinon to a compatible version with Internet Explorer and adds an ignore rule so Dependabot will not offer upgrades for this specific dependency (closes #7734).

The tests for Internet Explorer seem to fail, so we will have to open up another PR to fix these issues. @johnd0e perhaps you can provide some insights as to why these tests seem to be failing? It seems to be related to the `prosthetic-hand` dependency.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,7 +95,9 @@
     },
     "overrides": [
       {
-        "files": ["build/**/*"],
+        "files": [
+          "build/**/*"
+        ],
```
Comment (jonkoops) [inline package.json:100]: NPM keeps formatting the file like this when the dependencies change, so I included it here as well.

**Comment by johnd0e** [comment]
> @johnd0e perhaps you can provide some insights as to why these tests seem to be failing?

Sure, will check.
(perhaps the tests requires some minor changes after #7029)

**Comment by johnd0e** [comment]
Ref: #7414

- 2 tests ("in CSS scaled container") fail because Leaflet really fail them and should be fixed (should it?)
- 5 tests ("Object doesn't support this action") require `"prosthetic-hand": "^1.4.0"` (which is not published on npm)

@morner?

**Comment by mourner** [comment]
@IvanSanchez I don't have permissions to publish to NPM — can you do `npm publish` in `prosthetic-hand`, and then add more owners?

**Comment by mourner** [comment]
OK published prosthetic-hand v1.4.0 now, let's try that out!

**Comment by jonkoops** [comment]
Bumped the version in #7745 and rebased the PR.

**Comment by mourner** [comment]
@johnd0e looks like most failures are related to recent changes to `on` / `off` / etc., although not sure why only IE11 fails there.

**Comment by johnd0e** [comment]
That is because `false._leaflet_id=1` is ok in IE, but expected to fail in tests

**Review by johnd0e — APPROVED** [review]

**Pull request merged**